google/web-stories-wp
# Video Muting: Fix resource state updates

## Context



We got a user report about issues with resource updates when muting a video. The "Remove audio" button was simply not going away, yet the new file was uploading successfully.

## Summary



The reason for this bug:

While the `isMuting`/`isMuted` flags were properly set during start and progress, they got reset again at the end of the upload process? `additionalData.is_muted` was set to `true` when uploading the file to WordPress, yet the REST API response contained `is_muted: false` because the meta value was not properly updated

This PR ensures that the flags are kept and that the "Remove audio" button is correctly disappearing after muting.

## Relevant Technical Choices



* See individual commits for details
* Updates `updateExistingElements` to actually update the current state and only update the resource properties that are needed, so no data is potentially lost
* Adding some todos for followup tickets.
* Code simplification
* Prefixes the REST API field

## To-do



N/A

## User-facing changes



https://user-images.githubusercontent.com/841956/136575412-938a7afa-826d-471b-adba-3e6687e57bae.mov


## Testing Instructions






This PR can be tested by following these steps:

1. Add a new video with sound to the page
2. Click on "Remove audio"
3. See that new video is uploading
4. After upload finishes, see that "Remove audio" button is removed

Also:

* Verify that video trimming and video optimization via the design panel still works.


## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #9335
